fast64_internal/f3d/f3d_material.py

```diff
@@ -1271,467 +1334,380 @@ def update_node_values_of_material(material, context):
-        (texProperty.S.high + 1) / tex_size[0],
-        (texProperty.T.high + 1) / tex_size[1],
-        0,
-    )
+    # S/T High
+    inputs[str_index + " S High"].default_value = round_10_2(texProperty.S.high)
+    inputs[str_index + " T High"].default_value = round_10_2(texProperty.T.high)
 
     # Clamp
-    isTexGen = self.f3d_mat.rdp_settings.g_tex_gen or self.f3d_mat.rdp_settings.g_tex_gen_linear
-    nodes["Get UV " + str(texIndex) + " F3D v3"].inputs[5].default_value = (
-        1 if texProperty.S.clamp and not isTexGen else 0,
-        1 if texProperty.T.clamp and not isTexGen else 0,
-        0,
-    )
+    inputs[str_index + " ClampX"].default_value = 1 if texProperty.S.clamp else 0
+    inputs[str_index + " ClampY"].default_value = 1 if texProperty.T.clamp else 0
 
-    # Normalized Mask
-    nodes["Get UV " + str(texIndex) + " F3D v3"].inputs[6].default_value = (
-        (2**texProperty.S.mask) / tex_size[0] if texProperty.S.mask > 0 else 0,
-        (2**texProperty.T.mask) / tex_size[1] if texProperty.T.mask > 0 else 0,
-        0,
-    )
+    # Mask
+    inputs[str_index + " S Mask"].default_value = texProperty.S.mask
+    inputs[str_index + " T Mask"].default_value = texProperty.T.mask
 
     # Mirror
-    nodes["Get UV " + str(texIndex) + " F3D v3"].inputs[7].default_value = (
-        1 if texProperty.S.mirror else 0,
-        1 if texProperty.T.mirror else 0,
-        0,
-    )
+    inputs[str_index + " MirrorX"].default_value = 1 if texProperty.S.mirror > 0 else 0
+    inputs[str_index + " MirrorY"].default_value = 1 if texProperty.T.mirror > 0 else 0
 
     # Shift
-    nodes["Get UV " + str(texIndex) + " F3D v3"].inputs[8].default_value = (texProperty.S.shift, texProperty.T.shift, 0)
+    inputs[str_index + " S Shift"].default_value = texProperty.S.shift
+    inputs[str_index + " T Shift"].default_value = texProperty.T.shift
 
-    # Scale
-    nodes["Get UV " + str(texIndex) + " F3D v3"].inputs[9].default_value = (
-        scale[0] * uvBasisScale[0],
-        scale[1] * uvBasisScale[1],
-        0,
-    )
 
-    # Normalized Half Pixel
-    nodes["Get UV " + str(texIndex) + " F3D v3"].inputs[10].default_value = (
-        1 / (2 * tex_size[0]),
-        1 / (2 * tex_size[1]),
-        0,
-    )
+def iter_tex_nodes(node_tree: bpy.types.NodeTree, texIndex: int) -> Generator[bpy.types.TextureNodeImage, None, None]:
+    for i in range(1, 5):
+        nodeName = f"Tex{texIndex}_{i}"
+        if node_tree.nodes.get(nodeName):
+            yield node_tree.nodes[nodeName]
 
 
-def update_tex_values_index(
-    self, context, texProperty, texNodeName, uvNodeName, isTexGen, uvBasisScale, scale, texIndex
-):
+def set_texture_nodes_settings(
+    material: bpy.types.Material, texProperty: "TextureProperty", texIndex: int
+) -> (list[int] | None):
+    node_tree = material.node_tree
+    f3dMat: "F3DMaterialProperty" = material.f3d_mat
+
+    # Return value
+    texSize: None | list["int"] = None
```

Comment:
```suggestion
    texSize: None | list[int] = None
```
also `Optional[list[int]]` may be adequate (`from typing import Optional`)

Reply:
ahhhh `Optional[list[int]]` is nice, updated it to use that instead